var/spack/repos/builtin/packages/vasp/package.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+from llnl.util import tty
+import os
+
+
+class Vasp(MakefilePackage):
+    """
+    The Vienna Ab initio Simulation Package (VASP)
+    is a computer program for atomic scale materials modelling,
+    e.g. electronic structure calculations
+    and quantum-mechanical molecular dynamics, from first principles.
+    """
+
+    homepage = "http://vasp.at"
+    url      = "file://{0}/vasp.5.4.4.tar.gz".format(os.getcwd())
+
+    version('5.4.4', sha256='5bd2449462386f01e575f9adf629c08cb03a13142806ffb6a71309ca4431cfb3') # noqa
+
+    resource(name='vaspsol',
+             git='https://github.com/henniggroup/VASPsol.git',
+             tag='V1.0',
+             when='+vaspsol')
+
+    variant('scalapack', default=False,
+            description='Enables build with SCALAPACK')
+
+    variant('cuda', default=False,
+            description='Enables running on Nvidia GPUs')
+
+    variant('vaspsol', default=False,
+            description='Enable VASPsol implicit solvation model\n'
+            'https://github.com/henniggroup/VASPsol')
+
+    depends_on('rsync', type='build')
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('fftw')
+    depends_on('mpi', type=('build', 'link', 'run'))
+    depends_on('netlib-scalapack', when='+scalapack')
+    depends_on('cuda', when='+cuda')
+
+    conflicts('+vaspsol', when='+cuda', msg='+vaspsol only available for CPU')
+
+    parallel = False
+
+    def edit(self, spec, prefix):
+        spec = self.spec
+
+        if '%gcc' in spec:
+            make_include = join_path('arch', 'makefile.include.linux_gnu')
+        else:
+            make_include = join_path('arch',
+                                     'makefile.include.linux_' +
+                                     spec.compiler.name)
+
+        os.rename(make_include, 'makefile.include')
+
+        # This bunch of 'filter_file()' is to make these options settable
+        # as environment variables
```

Comment:
I wonder if it isn't easier to just directly set these instead of using environment variables.

Reply:
This was due to a workmate needing to use `spack install --dirty ...` due to time constraints to build vasp at a customer site.